### PR TITLE
added solution of hw-6.2

### PR DESCRIPTION
**Pull request by KseniiaMakarova**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,27 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Rectangle other = (Rectangle) object;
+        return getWidth().equals(other.getWidth())
+                && getLength().equals(other.getLength())
+                && getColor().equals(other.getColor());
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: do we need to use getters here?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,27 @@ public Integer getLength() {
...
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 13;
+        result = result * 31 + getWidth().hashCode();
+        result = result * 31 + getLength().hashCode();
+        result = result * 31 + getColor().hashCode();
+        return result;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: what if field would be null?

Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Rectangle.java]: NPE :(
fixed this)

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Rectangle.java]: no - we are in the same class
fixed)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**